test/e2e/apimachinery/openapiv3.go

```diff
@@ -163,5 +173,17 @@ var _ = SIGDescribe("OpenAPIV3", func() {
 			diff := cmp.Diff(*openAPISpec, spec2)
 			framework.Failf("%s", diff)
 		}
+
+		cleanupSampleAPIServer(ctx, f.ClientSet, aggrclient, names, "v1beta1.wardle.example.com")
```

Comment:
Line 153 `defer cleanupSampleAPIServer(ctx, f.ClientSet, aggrclient, names, "v1beta1.wardle.example.com")` shouldn't be needed anymore since it's moved here?

Reply:
Do we need to ensure the resources created by the test get cleaned up regardless of success or failure? I kept the defer statement for that, otherwise the resources would be left over if `ExpectNoError` above fails and terminates the test.

Reply:
Ah good catch. What about this entire section in a defer statement? I'd just like to avoid calling the function twice.

Reply:
sure, done